tests/load/pipeline/test_merge_disposition.py

```diff
@@ -1059,34 +1202,32 @@ def r():
     with pytest.raises(DestinationCapabilitiesException):
         p.run(r())
 
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_sql_configs=True,
+        local_filesystem_configs=True,
+        supports_merge=True,
+        subset=["postgres", "filesystem"],  # test one SQL and one non-SQL destination
+    ),
+    ids=lambda x: x.name,
+)
+def test_upsert_merge_strategy_config(destination_config: DestinationTestConfiguration) -> None:
+    if destination_config.destination == "filesystem":
+        # TODO: implement validation and remove this test exception
```

Comment:
what do you mean with "validation"?

Reply:
Validation of user-provided configuration.

For example, we want to raise if `upsert` is chosen as merge strategy, but no `primary_key` has been defined.

For SQL destinations this happens in `verify_sql_job_client_schema`. We need similar checks for the `delta` route. Ideally we use the same code for both.